### PR TITLE
add create_wasm_array_with_string api support about stringref.

### PR DESCRIPTION
**Pull request by LevelCA**

### CODE REVIEW

File: runtime-library/utils/type_utils.c [hunk 1]
```diff
@@ -283,14 +283,97 @@ get_closure_struct_type(wasm_module_t wasm_module,
...
+    wasm_value_t val = { 0 };
+    val.gc_obj = NULL;
+    wasm_array_obj_t new_arr = wasm_array_obj_new_with_type(
+        exec_env, stringref_array_type, arrlen, &val);
+    /* Push object to local ref to avoid being freed at next allocation */
+    wasm_runtime_push_local_object_ref(exec_env, &local_ref);
+    local_ref.val = (wasm_obj_t)new_arr;
+
+    if (!new_arr) {
+        wasm_runtime_pop_local_object_ref(exec_env);
```
Comment (xujuntwt95329) [inline runtime-library/utils/type_utils.c:347]: Had better call `wasm_runtime_push_local_object_ref` after checking whether `new_arr` is null, then you don't need to pop it here.

File: runtime-library/utils/type_utils.c [hunk 2]
```diff
@@ -283,14 +283,97 @@ get_closure_struct_type(wasm_module_t wasm_module,
...
+    wasm_array_type_t stringref_array_type = NULL;
+
+    uint32_t res_arr_type_idx =
+        get_stringref_array_type(module, &stringref_array_type);
+
+    wasm_struct_type_t res_arr_struct_type = NULL;
+
+    /* get result array struct type */
+    get_array_struct_type(module, res_arr_type_idx, &res_arr_struct_type);
+
```
Comment (xujuntwt95329) [inline runtime-library/utils/type_utils.c:333]: Had better add 
``` C
bh_assert(stringref_array_type);
bh_assert(res_arr_struct_type);
```

File: runtime-library/utils/type_utils.c (hunk 1 above)
Comment (LevelCA) [reply runtime-library/utils/type_utils.c]: OK

File: runtime-library/utils/type_utils.c (hunk 2 above)
Comment (LevelCA) [reply runtime-library/utils/type_utils.c:336]: done

File: runtime-library/utils/type_utils.c [hunk 3]
```diff
@@ -283,14 +283,103 @@ get_closure_struct_type(wasm_module_t wasm_module,
...
+        wasm_array_obj_set_elem(new_arr, i, &val);
+    }
+
+    wasm_struct_obj_t new_stringref_array_struct =
+        wasm_struct_obj_new_with_type(exec_env, res_arr_struct_type);
+
+    if (!new_stringref_array_struct) {
+        wasm_runtime_set_exception(wasm_runtime_get_module_inst(exec_env),
+                                   "alloc memory failed");
+        return NULL;
```
Comment (xujuntwt95329) [inline runtime-library/utils/type_utils.c:372]: Shoud `wasm_runtime_pop_local_object_ref(exec_env);` before return

Comment (LevelCA) [reply runtime-library/utils/type_utils.c:373]: done

**Pull request merged**